sdc/datatypes/hpat_pandas_dataframe_rolling_functions.py

```diff
@@ -78,45 +82,149 @@
+    rolling_params = df_rolling_params_codegen()
+    method_kws = {k: k for k in kwargs}
+    impl_params = ['self'] + args + params2list(kwargs)
     impl_params_as_str = ', '.join(impl_params)
 
+    data_columns = {col: idx for idx, col in enumerate(self.data.columns)}
+    other_columns = {col: idx for idx, col in enumerate(other.columns)}
+
+    # columns order matters
+    common_columns = [col for col in data_columns if col in other_columns]
+    all_columns = [col for col in data_columns]
+    for col in other_columns:
+        if col in all_columns:
+            continue
+        all_columns.append(col)
+
     results = []
-    impl_name = f'_df_rolling_{method_name}_impl'
+    impl_name = f'_df_rolling_{method_name}_with_other_df_impl'
     func_lines = [f'def {impl_name}({impl_params_as_str}):']
 
-    for idx, col in enumerate(self.data.columns):
+    if 'pairwise' in kwargs:
+        func_lines += [
+            '  if pairwise is None:',
+            '    _pairwise = False',
+            '  else:',
+            '    _pairwise = pairwise',
+            '  if _pairwise:',
+            '    raise ValueError("Method rolling.corr(). The object pairwise\\n expected: False, None")'
+        ]
+
+    data_length = 'len(get_dataframe_data(self._data, 0))' if data_columns else '0'
+    other_length = 'len(get_dataframe_data(other, 0))' if other_columns else '0'
+    func_lines += [f'  length = max([{data_length}, {other_length}])']
+
+    for col in all_columns:
+        res_data = f'result_data_{col}'
+        if col in common_columns:
+            other_series = f'other_series_{col}'
+            method_kws['other'] = other_series
+            method_params = ', '.join(args + params2list(method_kws))
+            func_lines += [
+                f'  data_{col} = get_dataframe_data(self._data, {data_columns[col]})',
+                f'  other_data_{col} = get_dataframe_data(other, {other_columns[col]})',
+                f'  series_{col} = pandas.Series(data_{col})',
+                f'  {other_series} = pandas.Series(other_data_{col})',
+                f'  rolling_{col} = series_{col}.rolling({rolling_params})',
+                f'  result_{col} = rolling_{col}.corr({method_params})',
+                f'  {res_data} = result_{col}._data[:length]'
+            ]
+        else:
+            func_lines += [
+                f'  {res_data} = numpy.empty(length, dtype=float64)',
+                f'  {res_data}[:] = numpy.nan'
+            ]
+        results.append((col, res_data))
+
+    data = ', '.join(f'"{col}": {data}' for col, data in results)
+    func_lines += [f'  return pandas.DataFrame({{{data}}})']
+    func_text = '\n'.join(func_lines)
+
+    global_vars = {'numpy': numpy, 'pandas': pandas, 'float64': float64,
+                   'get_dataframe_data': get_dataframe_data}
+
+    return func_text, global_vars
+
+
+def df_rolling_method_tail_codegen(method_params, df_columns, method_name):
```

Comment:
@densmirn I think this function can really be incorporated into df_rolling_method_codegen, since it's used only once now. So that df_rolling_method_codegen and df_rolling_method_with_other_df_codegen have similar look.